source/extensions/transport_sockets/alts/tsi_socket.cc

```diff
@@ -106,14 +106,22 @@ Network::PostIoAction TsiSocket::doHandshakeNextDone(NextResultPtr&& next_result
     }
     if (handshake_validator_) {
       std::string err;
-      const bool peer_validated = handshake_validator_(peer, err);
+      TsiInfo tsi_info;
+      const bool peer_validated = handshake_validator_(peer, tsi_info, err);
       if (peer_validated) {
         ENVOY_CONN_LOG(debug, "TSI: Handshake validation succeeded.", callbacks_->connection());
       } else {
         ENVOY_CONN_LOG(debug, "TSI: Handshake validation failed: {}", callbacks_->connection(),
                        err);
         return Network::PostIoAction::Close;
       }
+      ProtobufWkt::Struct dynamic_metadata;
+      ProtobufWkt::Value val;
+      val.set_string_value(tsi_info.name_);
+      dynamic_metadata.mutable_fields()->insert({std::string("peer_identity"), val});
+      callbacks_->connection().streamInfo().setDynamicMetadata("envoy.transport_sockets.alts",
```

Comment:
I think the metadata key can be more generic, such as `envoy.transport_socket.peer_information`. For instance peer identity can also be set by TLS transport with mTLS support.